### PR TITLE
Addresses config change in lxc 3

### PR DESCRIPTION
**Pull request by battlemidget**
Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>

### CODE REVIEW

**Comment by Cynerva** [comment]
LGTM

File: canonical-kubernetes/steps/00_process-providertype/before-deploy [hunk 1]
```diff
@@ -6,7 +6,13 @@ set -eux
...
 if [[ "$JUJU_PROVIDERTYPE" == "localhost" ]]; then
     debug "Running pre-deploy for $CONJURE_UP_SPELL"
-    sed "s/##MODEL##/$JUJU_MODEL/" "$(scriptPath)/lxd-profile.yaml" | lxc profile edit "juju-$JUJU_MODEL"
+    lxc_aa_profile="lxc.aa_profile"
+    version=$(lxc --version)
+    version_arr=(${version//./})
+    if [[ "${version_arr[0]}" -gt "2" ]]; then
+        lxc_aa_profile="lxc.apparmor.profile"
+    fi
+    cat "$(scriptPath)/lxd-profile.yaml" | sed "s/##MODEL##/$JUJU_MODEL/"  | sed "s/##AA_PROFILE##/$lxc_aa_profile/" | lxc profile edit "juju-$JUJU_MODEL"
```
Comment (kwmonroe) [inline canonical-kubernetes/steps/00_process-providertype/before-deploy]: this sed bad. make sed good:

```diff
- cat "$(scriptPath)/lxd-profile.yaml" | sed "s/##MODEL##/$JUJU_MODEL/"  | sed "s/##AA_PROFILE##/$lxc_aa_profile/" | lxc profile edit "juju-$JUJU_MODEL"
+ cat "$(scriptPath)/lxd-profile.yaml" | sed -e "s/##MODEL##/$JUJU_MODEL/" -e "s/##AA_PROFILE##/$lxc_aa_profile/" | lxc profile edit "juju-$JUJU_MODEL"
```

File: canonical-kubernetes/steps/00_process-providertype/before-deploy [hunk 2]
```diff
@@ -6,7 +6,13 @@ set -eux
 
 if [[ "$JUJU_PROVIDERTYPE" == "localhost" ]]; then
     debug "Running pre-deploy for $CONJURE_UP_SPELL"
-    sed "s/##MODEL##/$JUJU_MODEL/" "$(scriptPath)/lxd-profile.yaml" | lxc profile edit "juju-$JUJU_MODEL"
+    lxc_aa_profile="lxc.aa_profile"
+    version=$(lxc --version)
+    version_arr=(${version//./})
```
Comment (kwmonroe) [inline canonical-kubernetes/steps/00_process-providertype/before-deploy]: I don't think this is what you want to do, as this removes all `.` from the `${version}`:

```bash
$ version=$(lxc --version)
$ version_arr=(${version//./})
$ echo ${version_arr[0]}
2011
```
As you know, 2011 is greater than 2, so you'll always fall into the condition below.

Comment (kwmonroe) [reply canonical-kubernetes/steps/00_process-providertype/before-deploy]: the missing piece here is the space when constructing the `version_arr`:

```diff
- version_arr=(${version//./})
+ version_arr=(${version//./ })
```
Or use IFS and moar wizardry.

**Pull request merged**